zyte_api/_sync.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+from tenacity import AsyncRetrying
+from typing import Generator, List, Optional, Union
+
+from aiohttp import ClientSession
+
+from .aio.client import AsyncClient
+from .constants import API_URL
+
+
+class ZyteAPI:
+    """Synchronous Zyte API client.
+
+    To create an instance, pass your API key:
+
+    .. code-block:: python
+
+        client = ZyteAPI(api_key="YOUR_API_KEY")
+
+    Or :ref:`use an environment variable <api-key>` and omit your API key:
+
+    .. code-block:: python
+
+        client = ZyteAPI()
+
+    Use :meth:`get` and :meth:`iter` to send queries to Zyte API.
+    """
+
+    def __init__(
+        self,
+        *,
+        api_key=None,
+        api_url=API_URL,
+        n_conn=15,
+        retrying: Optional[AsyncRetrying] = None,
+        user_agent: Optional[str] = None,
+    ):
+        self._async_client = AsyncClient(
+            api_key=api_key,
+            api_url=api_url,
+            n_conn=n_conn,
+            retrying=retrying,
+            user_agent=user_agent,
+        )
+
+    def get(
+        self,
+        query: dict,
+        *,
+        endpoint: str = 'extract',
+        session: Optional[ClientSession] = None,
+        handle_retries: bool = True,
+        retrying: Optional[AsyncRetrying] = None,
+    ) -> dict:
+        """Send a query to Zyte API and get the result.
+
+        .. code-block:: python
+
+            result = client.get({"url": "https://toscrape.com", "httpResponseBody": True})
+        """
+        return asyncio.run(
+            self._async_client.request_raw(
+                query=query,
+                endpoint=endpoint,
+                session=session,
+                handle_retries=handle_retries,
+                retrying=retrying,
+            )
+        )
+
+
+    def iter(
```

Comment:
@Gallaecio that's such a nice API! What happens with errors, do you think we should document them if they can happen? I recall from reading the async batch example that part really worried me.

Reply:
It’s a would point, they probably raise. I’ll cover it in tests, and maybe figure out how to handle them nicely (e.g. maybe an option to either return them, raise on the first error, or raise the first error after all non-errors have been iterated).

Reply:
@Gallaecio hmm good question. Does exception allow access to the response? Do we raise exceptions for all https://docs.zyte.com/zyte-api/usage/errors.html#unsuccessful-responses or it's only for auth failures and connectivity issues?

Reply:
[We raise exceptions for all ≥400 responses](https://github.com/zytedata/python-zyte-api/blob/931a8103b5dfc63f40be4e663306600c440b7fa7/zyte_api/aio/client.py#L93). In those cases, [the exception does allow access to the response](https://github.com/zytedata/python-zyte-api/blob/931a8103b5dfc63f40be4e663306600c440b7fa7/zyte_api/aio/client.py#L105).

Reply:
Nice, then returning an exception makes sense, thank you 👍